core/data-transformers/wishlists-transformer.ts

```diff
@@ -55,18 +55,27 @@ function wishlistItemsTransformer(
   formatter: ExistingResultType<typeof getFormatter>,
   pt?: ExistingResultType<typeof getTranslations<'Product.ProductDetails'>>,
 ): WishlistItem[] {
-  return removeEdgesAndNodes(wishlistItems).map((item) => ({
```

Comment:
Can we just use `.filter().map()` here instead?

Reply:
@chanceaclark That's what I had originally, but for some reason typescript was having a hard time inferencing that `item.product` is not null after that operation, so I went with reduce since it's easier for TS to infer the type with an early return

Reply:
You can provide a type-guard to the `.filter` to narrow the returned type so it should be doable.

Reply:
@chanceaclark True, I sometimes forget those exist. It doesn't look pretty, but it works. Updated!

Reply:
It's fine if it doesn't look super pretty, we just prefer it as it's easier to comprehend `.filter.map` over `.reduce` most of the times. And we don't necessarily need to worry about "efficiency" here in the JS land as the compiler and engine will handle some of these refinements.